integration-tests/tests/run-workflow.js

```diff
@@ -28,38 +42,54 @@ const testRunWorkflowFn = _.flow(
   await click(page, findWorkflowButton)
   await click(page, clickable({ textContains: workflowName }))
   await waitForNoSpinners(page)
-  await click(page, clickable({ text: 'Add to Workspace' }))
   // note that this automatically brings in the highest numbered config, which isn't what happens when going through the method repo in FC
+  await click(page, clickable({ text: 'Add to Workspace' }))
+  // If we get this far, we need to ensure that we cleanup in case this workspace is going to be reused, thus the try, finally.
+  try {
+    await waitForNoSpinners(page)
+    await click(page, clickable({ text: 'Select Data' }))
+    await click(page, input({ labelContains: 'Choose specific test_entitys to process' }))
+    await click(page, `//*[@role="checkbox" and contains(@aria-label, "${testEntity.name}")]`)
+    await click(page, clickable({ text: 'OK' }))
+    await click(page, clickable({ text: 'Run analysis' }))
 
-  await waitForNoSpinners(page)
-  await click(page, clickable({ text: 'Select Data' }))
-  await click(page, input({ labelContains: 'Choose specific test_entitys to process' }))
-  await click(page, `//*[@role="checkbox" and contains(@aria-label, "${testEntity.name}")]`)
-  await click(page, clickable({ text: 'OK' }))
-  await click(page, clickable({ text: 'Run analysis' }))
-
-  await Promise.all([
-    page.waitForNavigation(),
-    click(page, clickable({ text: 'Launch' }))
-  ])
+    await Promise.all([
+      page.waitForNavigation(),
+      click(page, clickable({ text: 'Launch' }))
+    ])
 
-  await pRetry(async () => {
-    try {
-      await findInGrid(page, 'Succeeded', { timeout: 65 * 1000 }) // long enough for the submission details to refresh
-    } catch (e) {
-      throw new Error(e)
-    }
-  }, { retries: 10, factor: 1 })
+    await pRetry(async () => {
+      try {
+        await findInGrid(page, 'Succeeded', { timeout: 65 * 1000 }) // long enough for the submission details to refresh
+      } catch (e) {
+        throw new Error(e)
+      }
+    }, { retries: 10, factor: 1 })
 
-  await click(page, navChild('data'))
-  await click(page, clickable({ textContains: 'test_entity' }))
-  await findInDataTableRow(page, testEntity.name, testEntity.attributes.input)
-})
+    await click(page, navChild('data'))
+    await click(page, clickable({ textContains: 'test_entity' }))
+    await findInDataTableRow(page, testEntity.name, testEntity.attributes.input)
+  } finally {
```

Comment:
(just curious) could we add a `if (!newEntity)` here? to skip some of this cleanup?

Reply:
Yes, I can add that. It might speed up the tests in some places.